### PR TITLE
Create ServiceBase-doesnt-propagate-OnStart-exceptions.md

### PR DESCRIPTION
**Pull request by AlexGhiondea**

### CODE REVIEW

File: Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+4.7.1
+
+### Source Analyzer Status
+
+NotPlanned
+
+### Change Description
+
+Exceptions from services that fail to start will now be propagated from ServiceBase.Run
```
Comment (rpetrusha) [inline Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md]: What is the previous (4.7 and earlier) behavior? No exception thrown? Also, is this an opt-in feature (you opt-out by default)? In that case:

In the .NET Framework 4.7 and earlier versions, exceptions thrown on service startup are not propagated to the caller of <xref::System.ServiceProcess.Servicebase.Run%2A>. 

Starting with applications that target the .NET Framework 4.7.1, the runtime propagates exceptions to  <xref::System.ServiceProcess.Servicebase.Run%2A> for services that fail to start.

File: Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+On service start, if there is an exception, that exception will be propagated. This should help diagnose cases where services fail to start.
```
Comment (rpetrusha) [inline Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md]: Add the following:

If this behavior is undesirable, you can opt out of it by adding the following [`<AppContextSwitchOverrides>`](~/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element to the `<runtime>` section of your application configuration file:

```xml
<AppContextSwitchOverrides value="Switch.System.ServiceProcess.DontThrowExceptionsOnStart=true" />
```

Also, can you opt into the behavior for applications that target previous versions of the .NET Framework?

File: Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+### Affected APIs
+* `M:System.ServiceProcess.ServiceBase.Run`
```
Comment (rpetrusha) [inline Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md]: Add the following:

File: Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md (hunk 2 above)
Comment (AlexGhiondea) [reply Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md]: Done

**Comment by AlexGhiondea** [comment]
@rpetrusha I updated the document.

**Comment by rpetrusha** [comment]
Thanks, @AlexGhiondea, this looks good except for one of the xrefs and two extra colons.

**Comment by AlexGhiondea** [comment]
@rpetrusha I think I got them right this time.

I could not figure out which ones you were referring to... so if you can take a look that would be great!

**Comment by rpetrusha** [comment]
I made a few changes, @AlexGhiondea. I'll review it again now, but I think that it's ready to merge.

**Comment by AlexGhiondea** [comment]
Thank you @rpetrusha

**Pull request merged**